Source/NSStoryboard.m

```diff
@@ -0,0 +1,975 @@
+                {
+                  NSXMLElement *objects = (NSXMLElement *)[appNode parent];
+                  NSArray *appConsArr = [appNode nodesForXPath: @"connections" error: NULL];
+                  NSXMLNode *appCons = [appConsArr objectAtIndex: 0];
+                  if (appCons != nil)
+                    {
+                      [appCons detach];
+                    }
+                  
+                  NSArray *appChildren = [appNode children];
+                  NSEnumerator *ace = [appChildren objectEnumerator];
+                  NSXMLElement *ae = nil;
+                  
+                  // Move all application children to objects...
+                  while ((ae = [ace nextObject]) != nil)
+                    {
+                      [ae detach];
+                      [objects addChild: ae];
+                    }
+                  
+                  // Remove the appNode
+                  [appNode detach];
+                  
+                  // create a customObject entry for NSApplication reference...
+                  NSXMLNode *appCustomClass = (NSXMLNode *)[(NSXMLElement *)appNode
+                                                               attributeForName: @"customClass"];
+                  customClassString = ([appCustomClass stringValue] == nil) ?
+                    @"NSApplication" : [appCustomClass stringValue];
+                  NSXMLElement *customObject = nil;
+                  
+                  customObject = 
+                    [self _createCustomObjectWithId: @"-3"
+                                          userLabel: @"Application"
+                                        customClass: @"NSObject"];
+                  [child insertChild: customObject
+                             atIndex: 0];
+                  customObject = 
+                    [self _createCustomObjectWithId: @"-1"
+                                          userLabel: @"First Responder"
+                                        customClass: @"FirstResponder"];
+                  [child insertChild: customObject
+                             atIndex: 0];
+                  customObject =
+                    [self _createCustomObjectWithId: @"-2"
+                                          userLabel: @"File's Owner"
+                                        customClass: customClassString];
+                  if (appCons != nil)
+                    {
+                      [customObject addChild: appCons];
+                    }
+                  [child insertChild: customObject
+                             atIndex: 0];
+                  
+
+                  // Add it to the document
+                  [objects detach];
+                  [doc addChild: objects];
+                  
+                  // Assign application scene...
+                  ASSIGN(_applicationSceneId, sceneId);
+                }
+              else
+                {
+                  NSXMLElement *customObject = nil;
```

Comment:
Most of the code in this branch is very similar to the if branch, wouldn't it be possible to share more of the code?

Reply:
I believe so.  I will try to do this now.

Reply:
I'll save this for the refactor as well.  Since I am planning on reducing and moving all of the code.